jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -202,6 +196,48 @@ def _active_step_changed(self, *args):
     def slice_plugin(self):
         return self.app._jdaviz_helper.plugins['Slice']
 
+    @observe('aperture_items')
+    @skip_if_not_tray_instance()
+    def _aperture_items_changed(self, msg):
+        if not hasattr(self, 'aperture'):
+            return
+        for item in msg['new']:
+            if item not in msg['old']:
+                if item.get('type') != 'spatial':
+                    continue
+                subset_lbl = item.get('label')
+                try:
+                    self._extract_in_new_instance(subset_lbl=subset_lbl,
+                                                  auto_update=True, add_data=True)
+                except Exception:
+                    msg = SnackbarMessage(
+                        f"Automatic spectrum extraction for {subset_lbl} failed",
+                        color='error', sender=self, timeout=10000)
+                    raise  # TODO: remove before merge
+                else:
+                    msg = SnackbarMessage(
+                        f"Automatic spectrum extraction for {subset_lbl} successful",
+                        color='success', sender=self)
+                self.app.hub.broadcast(msg)
+
+    def _extract_in_new_instance(self, dataset=None, function='Sum', subset_lbl=None,
+                                 auto_update=False, add_data=False):
+        # create a new instance of the Spectral Extraction plugin (to not affect the instance in
+        # the tray) and extract the entire cube with defaults.
+        if dataset is None:
+            if self._app._jdaviz_helper._loaded_flux_cube is None:
+                return
+            dataset = self._app._jdaviz_helper._loaded_flux_cube.label
+        plg = self.new()
```

Comment:
Is this new instance automatically garbage collected when this method is done? If not, will we end up with too many instances?

Is there a way to accomplish this without a hidden plugin instance, maybe via a simple utility function instead?

And for large cube, we might not even want to do this at all, or at least make it more performant via random sampling. (This is a different ticket.)

Reply:
There shouldn't be anything keep it alive, so I think this is safe.  It would be possible to allow all entries to have kwarg overrides, but then labeling logic, etc, gets messy as that also needs to pass overrides and cannot respond to traitlet changes, etc.  If anyone can show a case where creating these clones causes memory issues, then that is our only choice - but if this works, I think its cleaner to follow and means we don't need to add a lot of extra logic to keep the plugin state and kwargs separate.